lib/common

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -ex
+
+export ALL_TARGET_ARCH=(amd64 arm32v6 arm64v8)
+export DEFAULT_ARCH=amd64
+
+GetCerbotVersionFromTag() {
+    TAG=$1
+    echo "${TAG//v/}"
+}
+
+GetQemuArch() {
+    ARCH=$1
+
+    case "$ARCH" in
+        "amd64")
+            echo "x86_64"
+            ;;
+        "arm32v6")
+            echo "arm"
+            ;;
+        "arm64v8")
+            echo "aarch64"
+            ;;
+        "*")
+            echo "Not supported build architecture '$1'." >&2
+            exit -1
+    esac
+}
+
+DownloadQemuStatic() {
+    ARCH=$1
+
+    QEMU_ARCH=$(GetQemuArch "$ARCH")
+    if [ ! -f "qemu-${QEMU_ARCH}-static" ]; then
+        QEMU_DOWNLOAD_URL="https://github.com/multiarch/qemu-user-static/releases/download"
+        QEMU_LATEST_TAG=$(curl -s https://api.github.com/repos/multiarch/qemu-user-static/tags \
+            | grep 'name.*v[0-9]' \
+            | head -n 1 \
+            | cut -d '"' -f 4)
+        curl -SL "${QEMU_DOWNLOAD_URL}/${QEMU_LATEST_TAG}/x86_64_qemu-$QEMU_ARCH-static.tar.gz" \
+            | tar xzv
+    fi
+}
+
+RegisterQemuHandlers() {
+    docker run --rm --privileged multiarch/qemu-user-static:register --reset
+}
+
+TagDockerImage() {
+    ARCH=$1
+    VERSION=$2
+
+    docker tag "${DOCKER_REPO}:${ARCH}-v${VERSION}" "${DOCKER_REPO}:${ARCH}-latest"
+    docker push "${DOCKER_REPO}:${ARCH}-v${VERSION}"
```

Comment:
Typically `TagDockerImage` will tag several images and push them. Two things:
* could we update the name of the method to make clear that tags will also be pushed ?
* since `TagDockerImage` is invoked during the `post_build` phase, does it means that a second push is done uncessarily during the default `push` phase of DockerHub?

Reply:
- `PushAllArquitectureDockerImages` could be a better name.
- Yes, it does a second unnecessary push. In fact, the image will already be at registry. I think that a better solution could be to use custom `push` hook and push all architecture dependent images. I will work on it.

Reply:
Changes done in 1dc60dcceba2141527d11e8af14ee0a4674c90c2